### ISSUE TITLE
enable postcode search to shut

### ISSUE THREAD

**Issue by polly64** [tomalrussell/colouring-london#198]

**Comment by tomalrussell** [comment]
Typing ESC or clicking x should clear results.

**Comment by tomalrussell** [comment]
Could be a good first UI issue, scoped to the [SearchBox](https://github.com/tomalrussell/colouring-london/blob/master/app/src/frontend/search-box.js) React component

**Comment by martin-dj** [comment]
Search doesn't work on my dev environment for some reason, so a bit difficult to test a fix. But thought something like this might do the trick?

```

class SearchBox extends Component {
    constructor(props) {
        super(props);
        ...
        this.clearResults = this.clearResults.bind(this);
    }

    // Update search term
    handleChange(e) {
        this.setState({
            q: e.target.value
        });

       // Attempt to intercept the ‘clear’ icon and clear results list as well 
        if(e.target.value === "") {
           this.setState({
              results: []
           });
        }
    }

    // Clear the results list resulting from a search on ESC key press
    clearResults(e){
        if(e.keyCode === 27) {
           //ESC is pressed
           console.log('You pressed the escape key!');
           this.setState({
              results: []
           });
        }
    }

    componentDidMount(){
        document.addEventListener("keydown", this.clearResults, false);
    }

    componentWillUnmount(){
        document.removeEventListener("keydown", this.clearResults, false);
    }
...
}

```

**Comment by tomalrussell** [comment]
It can be even simpler with the `onKeyDown` handler - add to the top-level div:

```js
        return (
            <div className={`search-box ${this.props.is_building? "building" : ""}`}
                onKeyDown={this.handleKeyPress}>
                ...
            </div>
```

then handle (not forgetting to bind these in the constructor):

```js
    handleKeyPress(e){
        if(e.keyCode === 27) {
            //ESC is pressed
            this.clearQuery();
            this.clearResults();
        }
    }

    clearResults(){
        this.setState({
            results: []
        });
    }

    clearQuery(){
        this.setState({
            q: ""
        });
    }
```

Here's the events reference: https://reactjs.org/docs/events.html#keyboard-events

**Comment by tomalrussell** [comment]
As for search not working - do you have any data in the search table? 

Maybe something to add to the setup docs - here's the script to load Open Postcodes: https://github.com/tomalrussell/colouring-london/blob/master/etl/load_postcodes.sh

Requires [`ogr2ogr`](http://www.gdal.org/ogr2ogr.html) which you can `sudo apt get gdal-bin` to install.

**Comment by martin-dj** [comment]
Thanks Tom, got search working on my dev environment and have submitted fix to:
feature/clear_pc_search_results
and made pull request

**Comment by tomalrussell** [comment]
Closed by #229
